Documentation about Gradle Version

Using Gradle <= 4.8 will cause this Exception:

```
Caused by: java.lang.NoSuchMethodError: org.gradle.api.tasks.TaskContainer.register(Ljava/lang/String;Ljava/lang/Class;Lorg/gradle/api/Action;)Lorg/gradle/api/tasks/TaskProvider;
        at io.miret.etienne.gradle.sass.SassGradlePlugin.apply(SassGradlePlugin.java:30)
```

Please add the information about minimum Gradle version in your `README.md`.

--- a/README.md
+++ b/README.md
@@ -61,6 +61,9 @@ compileSass {
   // (Default to emit)
   noErrorCss ()
 
+  // Watch sass files in sourceDir for changes
+  watch ()
+
   // Source map style:
   //  - file: output source map in a separate file (default)
   //  - embed: embed source map in CSS

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ If the [War Plugin][3] is also applied, the generated CSS will also be added to
 the war artifact. 
 
 ## Requirements
-* Gradle >= 4.9
+* Gradle >= 6.0
 
 ## Usage
 

--- a/README.md
+++ b/README.md
@@ -26,7 +26,7 @@ Optionally, configure it for your needs (default values shown):
 ```groovy
 sass {
   // dart-sass version to use:
-  version = '1.23.7'
+  version = '1.24.4'
 
   // Directory where to install dart-sass:
   directory = file ("${rootDir}/.gradle/sass")

--- a/README.md
+++ b/README.md
@@ -8,6 +8,9 @@ to `build/sass`.
 If the [War Plugin][3] is also applied, the generated CSS will also be added to
 the war artifact. 
 
+## Requirements
+* Gradle >= 4.9
+
 ## Usage
 
 First, apply the plugin:
@@ -18,7 +21,7 @@ plugins {
 }
 ```
 
-Optionally, configure it for your needs (default values chown):
+Optionally, configure it for your needs (default values shown):
 
 ```groovy
 sass {

--- a/src/functionalTest/java/io/miret/etienne/gradle/sass/SassGradlePluginFunctionalTest.java
+++ b/src/functionalTest/java/io/miret/etienne/gradle/sass/SassGradlePluginFunctionalTest.java
@@ -122,6 +122,22 @@ class SassGradlePluginFunctionalTest {
   }
 
   @Test
+  void should_watch () throws IOException {
+    Path out = createExecutable();
+
+    GradleRunner.create ()
+        .withPluginClasspath ()
+        .withArguments ("watch")
+        .withProjectDir (projectDir.toFile ())
+        .build();
+
+    assertThat (out).hasContent (String.format (
+       "sass --style=expanded --watch --source-map-urls=relative %1$s/src/main/sass:%1$s/build/sass",
+       projectDir.toRealPath ()
+    ));
+  }
+
+  @Test
   void should_disable_charset_output () throws IOException {
     Path out = createExecutable ();
 

--- a/src/functionalTest/resources/io/miret/etienne/gradle/sass/build.gradle
+++ b/src/functionalTest/resources/io/miret/etienne/gradle/sass/build.gradle
@@ -20,6 +20,10 @@ task compileCompressed (type: CompileSass) {
   style = compressed
 }
 
+task watch (type: CompileSass) {
+  watch ()
+}
+
 task compileNoCharset (type: CompileSass) {
   noCharset ()
 }

--- a/src/main/java/io/miret/etienne/gradle/sass/CompileSass.java
+++ b/src/main/java/io/miret/etienne/gradle/sass/CompileSass.java
@@ -10,7 +10,10 @@ import org.gradle.api.tasks.InputFiles;
 import org.gradle.api.tasks.Internal;
 import org.gradle.api.tasks.OutputDirectory;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.workers.WorkQueue;
+import org.gradle.workers.WorkerExecutor;
 
+import javax.inject.Inject;
 import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
@@ -18,6 +21,8 @@ import java.util.List;
 import static java.util.stream.Collectors.toList;
 
 public class CompileSass extends DefaultTask {
+
+  private final WorkerExecutor workerExecutor;
 
   enum Style {
     expanded,
@@ -126,6 +131,12 @@ public class CompileSass extends DefaultTask {
     return SourceMapUrls.absolute;
   }
 
+  @Inject
+  public CompileSass (WorkerExecutor workerExecutor) {
+    super();
+    this.workerExecutor = workerExecutor;
+  }
+
   @TaskAction
   public void compileSass () {
     String command = Os.isFamily (Os.FAMILY_WINDOWS) ? "sass.bat" : "sass";
@@ -138,39 +149,18 @@ public class CompileSass extends DefaultTask {
         .resolve (command)
         .toFile ();
 
-    getProject ().exec (execSpec -> {
-      execSpec.executable (executable);
-
-      List<String> args = new ArrayList<> ();
-      loadPaths.stream ()
-          .map (File::toString)
-          .map ("--load-path="::concat)
-          .forEach (args::add);
-      args.add (String.format ("--style=%s", style));
-      if (watch) {
-        args.add("--watch");
-      }
-      if (!charset) {
-        args.add ("--no-charset");
-      }
-      if (!errorCss) {
-        args.add ("--no-error-css");
-      }
-      switch (sourceMap) {
-        case none:
-          args.add ("--no-source-map");
-          break;
-        case embed:
-          args.add ("--embed-source-map");
-          break;
-        default:
-          // nothing to do.
-      }
-      if (sourceMap != SourceMap.none) {
-        args.add (String.format ("--source-map-urls=%s", sourceMapUrls));
-      }
-      args.add (String.format ("%s:%s", sourceDir, outputDir));
-      execSpec.args (args);
+    WorkQueue workQueue = workerExecutor.noIsolation();
+    workQueue.submit(CompileSassWorkAction.class, compileSassWorkParameters -> {
+      compileSassWorkParameters.getExecutable ().set (executable);
+      compileSassWorkParameters.getLoadPaths ().setFrom (loadPaths);
+      compileSassWorkParameters.getOutputDir ().set (outputDir);
+      compileSassWorkParameters.getSourceDir ().set (sourceDir);
+      compileSassWorkParameters.getStyle ().set (style);
+      compileSassWorkParameters.getSourceMap ().set (sourceMap);
+      compileSassWorkParameters.getSourceMapUrls ().set (sourceMapUrls);
+      compileSassWorkParameters.getWatch ().set (watch);
+      compileSassWorkParameters.getCharset ().set (charset);
+      compileSassWorkParameters.getErrorCss ().set (errorCss);
     });
   }
 

--- a/src/main/java/io/miret/etienne/gradle/sass/CompileSass.java
+++ b/src/main/java/io/miret/etienne/gradle/sass/CompileSass.java
@@ -49,6 +49,9 @@ public class CompileSass extends DefaultTask {
   private Style style = Style.expanded;
 
   @Getter (onMethod_ = {@Input})
+  private boolean watch = false;
+
+  @Getter (onMethod_ = {@Input})
   private boolean charset = true;
 
   @Getter (onMethod_ = {@Input})
@@ -82,6 +85,10 @@ public class CompileSass extends DefaultTask {
 
   public void noErrorCss () {
     errorCss = false;
+  }
+
+  public void watch () {
+    watch = true;
   }
 
   @Internal
@@ -140,6 +147,9 @@ public class CompileSass extends DefaultTask {
           .map ("--load-path="::concat)
           .forEach (args::add);
       args.add (String.format ("--style=%s", style));
+      if (watch) {
+        args.add("--watch");
+      }
       if (!charset) {
         args.add ("--no-charset");
       }

--- a/src/main/java/io/miret/etienne/gradle/sass/CompileSassWorkAction.java
+++ b/src/main/java/io/miret/etienne/gradle/sass/CompileSassWorkAction.java
@@ -1,0 +1,59 @@
+package io.miret.etienne.gradle.sass;
+
+import org.gradle.process.ExecOperations;
+import org.gradle.workers.WorkAction;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class CompileSassWorkAction implements WorkAction<CompileSassWorkParameters> {
+  private final ExecOperations execOperations;
+
+  @Inject
+  public CompileSassWorkAction(ExecOperations execOperations) {
+    this.execOperations = execOperations;
+  }
+
+  @Override
+  public void execute() {
+    CompileSassWorkParameters parameters = getParameters();
+
+    execOperations.exec (execSpec -> {
+      execSpec.executable (parameters.getExecutable().get());
+
+      List<String> args = new ArrayList<>();
+      parameters.getLoadPaths ().getFiles ().stream ()
+          .map (File::toString)
+          .map ("--load-path="::concat)
+          .forEach (args::add);
+      args.add (String.format ("--style=%s", parameters.getStyle().get()));
+      if (parameters.getWatch().get()) {
+        args.add("--watch");
+      }
+      if (!parameters.getCharset().get()) {
+        args.add ("--no-charset");
+      }
+      if (!parameters.getErrorCss().get()) {
+        args.add ("--no-error-css");
+      }
+      CompileSass.SourceMap sourceMap = parameters.getSourceMap().get();
+      switch (sourceMap) {
+        case none:
+          args.add ("--no-source-map");
+          break;
+        case embed:
+          args.add ("--embed-source-map");
+          break;
+        default:
+          // nothing to do.
+      }
+      if (sourceMap != CompileSass.SourceMap.none) {
+        args.add (String.format ("--source-map-urls=%s", parameters.getSourceMapUrls().get()));
+      }
+      args.add (String.format ("%s:%s", parameters.getSourceDir().get(), parameters.getOutputDir().get()));
+      execSpec.args (args);
+    });
+  }
+}

--- a/src/main/java/io/miret/etienne/gradle/sass/CompileSassWorkParameters.java
+++ b/src/main/java/io/miret/etienne/gradle/sass/CompileSassWorkParameters.java
@@ -1,0 +1,22 @@
+package io.miret.etienne.gradle.sass;
+
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.workers.WorkParameters;
+
+public interface CompileSassWorkParameters extends WorkParameters {
+  RegularFileProperty getExecutable();
+  ConfigurableFileCollection getLoadPaths();
+
+  RegularFileProperty getOutputDir();
+  RegularFileProperty getSourceDir();
+
+  Property<CompileSass.Style> getStyle();
+  Property<CompileSass.SourceMap> getSourceMap();
+  Property<CompileSass.SourceMapUrls> getSourceMapUrls();
+
+  Property<Boolean> getWatch();
+  Property<Boolean> getCharset();
+  Property<Boolean> getErrorCss();
+}

--- a/src/main/java/io/miret/etienne/gradle/sass/SassGradlePluginExtension.java
+++ b/src/main/java/io/miret/etienne/gradle/sass/SassGradlePluginExtension.java
@@ -17,7 +17,7 @@ public class SassGradlePluginExtension {
   private String baseUrl;
 
   public SassGradlePluginExtension (Project project) {
-    this.version = "1.24.2";
+    this.version = "1.24.4";
     this.directory = project.getRootDir ()
         .toPath ()
         .resolve (".gradle/sass")

--- a/src/main/java/io/miret/etienne/gradle/sass/SassGradlePluginExtension.java
+++ b/src/main/java/io/miret/etienne/gradle/sass/SassGradlePluginExtension.java
@@ -17,7 +17,7 @@ public class SassGradlePluginExtension {
   private String baseUrl;
 
   public SassGradlePluginExtension (Project project) {
-    this.version = "1.23.7";
+    this.version = "1.24.2";
     this.directory = project.getRootDir ()
         .toPath ()
         .resolve (".gradle/sass")

